python/cudf/cudf/core/scalar.py

```diff
@@ -16,7 +17,50 @@
 )
 
 
-class Scalar(BinaryOperand):
+# Note that the metaclass below can easily be generalized for use with
+# other classes, if needed in the future. Simply replace the arguments
+# of the `__call__` method with `*args` and `**kwargs`. This will
+# result in additional overhead when constructing the cache key, as
+# unpacking *args and **kwargs is not cheap. See the discussion in GH
+# #11246 for more details.
```

Comment:
Perhaps link to the specific thread? A cursory search of the PR doesn't turn up this specific discussion, so it may not be that obvious to find.

Reply:
Fix'd